base/compiler/ssair/passes.jl

```diff
@@ -110,7 +110,8 @@ function compute_value_for_use(ir::IRCode, domtree::DomTree, allblocks::Vector{I
     end
 end
 
-function simple_walk(compact::IncrementalCompact, @nospecialize(defssa#=::AnySSAValue=#), pi_callback=(pi, idx)->false)
+function simple_walk(compact::IncrementalCompact, @nospecialize(defssa::AnySSAValue),
```

Comment:
Declaring a type on a `@nospecialize` argument is normally a bad idea, as it may severely penalize the call

Reply:
Would you mind elaborating it ?
`@nospecialize` influences dispatches only at this moment, and I think every call of this function will be statically resolved anyway. And so I don't think there is any other effects than now we can get `MethodError` for wrong `defssa` types.

Reply:
it removes it from the dispatch cache